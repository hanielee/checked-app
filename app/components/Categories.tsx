import Container from "./Container";
import CategoryCard from "./CategoryCard";

import { IoRestaurantOutline, IoBagHandleOutline, IoLaptopOutline } from "react-icons/io5";
import { BiDrink, BiHotel, BiSpa, BiParty} from "react-icons/bi";
import { GiHiking, GiIsland, GiModernCity} from "react-icons/gi";

import { TbYoga } from "react-icons/tb";

export const categories = [
    {
        label: 'Restaurant',
        icon: IoRestaurantOutline,
    },
    {
        label: 'Bar',
        icon: BiDrink,
    },
    {
        label: 'Activity',
        icon: BiParty,
    },
    {
        label: 'Spa',
        icon: BiSpa,
    },
    {
        label: 'Yoga',
        icon: TbYoga,
    },
    {
        label: 'Hike',
        icon: GiHiking,
    },
    {
        label: 'Mall',
        icon: IoBagHandleOutline,
    },
    {
        label: 'Co-Working Space',
        icon: IoLaptopOutline,
    },
    {
        label: 'Island',
        icon: GiIsland,
    },
    {
        label: 'City',
        icon: GiModernCity,
    },
    {
        label: 'Hotel',
        icon: BiHotel,
    }
]

const Categories = () => {
    return ( 
    <Container>
    <div className="flex flex-row items-start justify-center h-screen">
        <div className="pt-4 pb-8 flex flex-row items-center justify-between gap-2 overflow-x-auto">
            <div className="grid grid-cols-4 gap-2">
                {categories.map((item) => (
                    <CategoryCard
                    key={item.label}
                    label={item.label}
                    icon={item.icon} 
                    />
                ))}
            </div>
        </div>
    </div>
    </Container> 
    );
}
 
export default Categories;