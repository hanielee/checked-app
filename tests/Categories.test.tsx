import { render, screen } from '@testing-library/react';
import Categories, { categories } from '..//app/components/Categories';  
import '@testing-library/jest-dom'


jest.mock('..//app/components/CategoryCard', () => ({
    __esModule: true,
    default: ({ label, icon: Icon }: any) => (
      <div data-testid={`category-card-${label}`}>
        <Icon size={30} />
        <span>{label}</span>
      </div>
    ),
  }));
  
  describe('Categories Component', () => {
    it('should render all category cards correctly', () => {
      render(<Categories />);
  
      categories.forEach((category) => {
        const categoryElement = screen.getByTestId(`category-card-${category.label}`);
        expect(categoryElement).toBeInTheDocument();
        expect(categoryElement).toHaveTextContent(category.label);
      });
    });
  
    it('should display icons for each category card', () => {
      render(<Categories />);
  
      categories.forEach((category) => {
        const categoryIcon = screen.getByTestId(`category-card-${category.label}`).querySelector('svg');
        expect(categoryIcon).toBeInTheDocument();
      });
    });
  
    it('should match the snapshot', () => {
      const { asFragment } = render(<Categories />);
      expect(asFragment()).toMatchSnapshot();
    });
  });