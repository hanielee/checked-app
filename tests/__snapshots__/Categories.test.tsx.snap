// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="max-w-[2520px] mx-auto xl:px-20 md:px-10 sm: px-2 px-4"
    data-testid="container"
  >
    <div
      class="flex flex-row items-start justify-center h-screen"
    >
      <div
        class="pt-4 pb-8 flex flex-row items-center justify-between gap-2 overflow-x-auto"
      >
        <div
          class="grid grid-cols-4 gap-2"
        >
          <div
            data-testid="category-card-Restaurant"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M57.49 47.74l368.43 368.43a37.28 37.28 0 010 52.72h0a37.29 37.29 0 01-52.72 0l-90-91.55a32 32 0 01-9.2-22.43v-5.53a32 32 0 00-9.52-22.78l-11.62-10.73a32 32 0 00-29.8-7.44h0a48.53 48.53 0 01-46.56-12.63l-85.43-85.44C40.39 159.68 21.74 83.15 57.49 47.74z"
                fill="none"
                stroke-linejoin="round"
                stroke-width="32"
              />
              <path
                d="M400 32l-77.25 77.25A64 64 0 00304 154.51v14.86a16 16 0 01-4.69 11.32L288 192m32 32l11.31-11.31a16 16 0 0111.32-4.69h14.86a64 64 0 0045.26-18.75L480 112m-40-40l-80 80M200 368l-99.72 100.28a40 40 0 01-56.56 0h0a40 40 0 010-56.56L128 328"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="32"
              />
            </svg>
            <span>
              Restaurant
            </span>
          </div>
          <div
            data-testid="category-card-Bar"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.832 4.555A1 1 0 0 0 20 3H4a1 1 0 0 0-.832 1.554L11 16.303V20H8v2h8v-2h-3v-3.697l7.832-11.748zM12 14.197 8.535 9h6.93L12 14.197zM18.132 5l-1.333 2H7.201L5.868 5h12.264z"
              />
            </svg>
            <span>
              Bar
            </span>
          </div>
          <div
            data-testid="category-card-Activity"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 8.31c-.36-.41-.73-.82-1.12-1.21l-.29-.27.14-.12a3.15 3.15 0 0 0 .9-3.49A3.91 3.91 0 0 0 14 1v2a2 2 0 0 1 1.76 1c.17.4 0 .84-.47 1.31-.07.08-.15.13-.22.2-3-2.41-6.29-3.77-7.9-2.16a2.16 2.16 0 0 0-.41.59v.1l-.18.53-4.41 13.1A3.28 3.28 0 0 0 5.28 22a3.21 3.21 0 0 0 1-.17L20 17.28a1 1 0 0 0 .43-.31l.21-.18c1.43-1.44.51-4.21-1.41-6.9A6.63 6.63 0 0 1 23 9V7a8.44 8.44 0 0 0-5 1.31zM5.7 19.93a1.29 1.29 0 0 1-1.63-1.63l1.36-4.1a10.7 10.7 0 0 0 4.29 4.39zm7-2.33a8.87 8.87 0 0 1-6.3-6.29l1-3 .06.09c.11.22.25.45.39.68s.16.29.26.44.33.48.51.73.19.28.3.42.43.55.66.82l.29.35c.34.39.7.77 1.08 1.16s.68.64 1 1l.33.28.78.63.37.28c.28.2.55.4.83.58l.31.2c.36.22.72.43 1.07.61h.05zm6.51-2.23h-.06c-.69.38-3.56-.57-6.79-3.81-.34-.34-.66-.67-.95-1l-.29-.35-.53-.64-.29-.4c-.13-.19-.27-.37-.39-.55l-.26-.42-.29-.47c-.08-.14-.14-.27-.21-.4s-.15-.26-.21-.4a3.31 3.31 0 0 1-.14-.36c-.05-.13-.11-.26-.15-.38S8.6 6 8.57 5.88s-.05-.22-.07-.32a2.26 2.26 0 0 1 0-.26 1 1 0 0 1 0-.24l.11-.31c.36-.36 2.23 0 4.73 1.9A4.13 4.13 0 0 1 12 7v2a6.45 6.45 0 0 0 3-.94l.48.46c.42.42.81.85 1.18 1.28a5.32 5.32 0 0 0-.6 3.4l2-.39a3.57 3.57 0 0 1 0-1.12 11.3 11.3 0 0 1 .81 1.45c.56 1.32.52 2.06.34 2.23z"
              />
            </svg>
            <span>
              Activity
            </span>
          </div>
          <div
            data-testid="category-card-Spa"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.787 9.023c-.125.027-1.803.418-3.953 1.774-.323-1.567-1.279-4.501-4.108-7.485L12 2.546l-.726.767C8.435 6.308 7.483 9.25 7.163 10.827 5.005 9.448 3.34 9.052 3.218 9.024L2 8.752V10c0 7.29 3.925 12 10 12 5.981 0 10-4.822 10-12V8.758l-1.213.265zM8.999 12.038c.002-.033.152-3.1 3.001-6.532C14.814 8.906 14.999 12 15 12v.125a18.933 18.933 0 0 0-3.01 3.154 19.877 19.877 0 0 0-2.991-3.113v-.128zM12 20c-5.316 0-7.549-4.196-7.937-8.564 1.655.718 4.616 2.426 7.107 6.123l.841 1.249.825-1.26c2.426-3.708 5.425-5.411 7.096-6.122C19.534 15.654 17.304 20 12 20z"
              />
            </svg>
            <span>
              Spa
            </span>
          </div>
          <div
            data-testid="category-card-Yoga"
          >
            <svg
              fill="none"
              height="30"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M12 4m-1 0a1 1 0 1 0 2 0a1 1 0 1 0 -2 0"
              />
              <path
                d="M4 20h4l1.5 -3"
              />
              <path
                d="M17 20l-1 -5h-5l1 -7"
              />
              <path
                d="M4 10l4 -1l4 -1l4 1.5l4 1.5"
              />
            </svg>
            <span>
              Yoga
            </span>
          </div>
          <div
            data-testid="category-card-Hike"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M185 58.59c-15-.38-30 13.1-33.1 33.43-3.3 21.68 8.8 39.98 24.8 42.38 16 2.4 32.9-11.5 36.1-33.2 3.3-21.72-8.7-39.94-24.8-42.35-1-.15-2-.24-3-.26zm113.3 77.21l-17.2 5.2 8.3 28.1 18.9.5zm-144.6 7.6c-2.7 70.5-.4 140.9-3.6 202.5-15.8 35.8-36.7 70.2-50.38 99.2l25.68 12.4c20.8-33.7 42.2-68 55.2-102.9l11.3-63.4 53 45.9 16.2 88.1 29.3-5.8-17.6-100.6-58.1-53.6-9.3-86.4 48.1 31.3c26.7.9 45.1-1.9 69.6-5.7l1-16.3-70.8-2.1-46.5-42c-9.5 6.7-20.9 10-32.8 8.2-7.7-1.2-14.5-4.3-20.3-8.8zm-16.6 13l-40.47 11.7-5.96 85.5c13.53 6.7 31.03 11.6 43.23 3.4zm352.7 65.3l-95.7 78.1-32 51.9L324 222.5c-6.3.9-12.2 1.8-18 2.6l43.5 147.1-22.6 36.7-19.4 3.4 3.7 21.3-64.5 12.8-4.2-22.7-84.3 14.8c-9.5 15.6-17.1 27.9-26 42.2l-49.52-23.9-62.64 34.8H489.8z"
              />
            </svg>
            <span>
              Hike
            </span>
          </div>
          <div
            data-testid="category-card-Mall"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M80 176a16 16 0 00-16 16v216c0 30.24 25.76 56 56 56h272c30.24 0 56-24.51 56-54.75V192a16 16 0 00-16-16zm80 0v-32a96 96 0 0196-96h0a96 96 0 0196 96v32"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="32"
              />
              <path
                d="M160 224v16a96 96 0 0096 96h0a96 96 0 0096-96v-16"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="32"
              />
            </svg>
            <span>
              Mall
            </span>
          </div>
          <div
            data-testid="category-card-Co-Working Space"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="304"
                rx="32.14"
                ry="32.14"
                stroke-linejoin="round"
                stroke-width="32"
                width="416"
                x="48"
                y="96"
              />
              <path
                d="M16 416h480"
                stroke-linecap="round"
                stroke-miterlimit="10"
                stroke-width="32"
              />
            </svg>
            <span>
              Co-Working Space
            </span>
          </div>
          <div
            data-testid="category-card-Island"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M80.87 28.32c-10.027.162-20.065 3.47-29.706 11.055C79.26 31.458 116.008 60.67 128.582 94.5c-33.088 2.865-77.492 21.193-92.373 60.79 45.182-35.396 77.437-49.508 97.192-28.644-20.36 20.232-37.693 49.855-34.722 77.06 8.497-19.502 30.642-47.206 53.763-56.956-.017.246-.03.493-.037.74 0 9.698 7.86 17.56 17.56 17.56 5.507-.01 10.692-2.603 14.003-7.005 43.084 62.306 46.703 103.963 46.99 171.13 1.22 3.765 3.31 13.657 8.712 13.323 2.138-.15 7.886-4.198 9.24-14.906-.658-72.08-6.662-120.87-59.648-192.89.088-.437.17-.88.244-1.335 12.77-25.514 63.138-12.534 85.207-7.342-19.952-24.276-63.064-33.383-91.26-30.154 6.987-23.99 41.58-35.786 79.522-39.88-35.283-14.532-83.623-2.6-108.498 18.582-18.92-23.63-46.22-46.692-73.61-46.252zM316.444 88.3c-14.417-.27-30.606 5.297-47.838 19.68 55.587-9.758 66.225 13.936 65.26 41.247-27.864-3.965-65.48 2.288-83.724 24.488 15.247-3.588 43.993-5.876 64.527 1.6-2.136 2.976-3.288 6.547-3.293 10.21 0 9.697 7.86 17.558 17.557 17.56 1.158-.007 2.312-.127 3.447-.36-29.184 40.13-43.586 77.41-49.65 109.765 5.913.638 11.845 1.472 17.78 2.49 6.267-33.1 22.157-72.1 56.822-115.246 3.327 3.807 8.134 5.994 13.19 6.002 8.54-.01 15.833-6.162 17.283-14.578 24.362 2.404 52.773 19.613 66.91 34.192-6.48-25.342-31.1-46.236-56.117-58.325 20.007-20.112 64.557-27.84 85.123-26.85-48.212-22.24-87.34-20.276-110.062-9.238-9.94-21.647-30.544-42.133-57.213-42.636zM18 327v18h100.234c14.542-6.786 29.8-12.894 45.434-18zm330.69 0c15.736 5.106 31.102 11.213 45.736 18H494v-18zm-81.858 2.29c-1.966 17.012-11.84 30.178-25.898 31.165-17.093-1.086-24.48-13.605-27.6-27.437-33.38 5.94-67.274 18.015-97.31 33.033-36.807 18.405-67.758 41.478-84.942 61.233 4.887 1.483 10.322 3.123 17 4.844 16.234 4.183 36.103 7.82 47.176 6.904 8.815-.73 18.05-5.583 28.39-11.27 10.34-5.687 21.82-12.22 35.834-13.026 19-1.092 36.012 5.71 51.84 12.04 15.828 6.332 30.557 12.207 44.69 12.226 8.875.012 18.36-3.293 28.83-7.22 10.47-3.925 21.902-8.468 34.943-8.778 30.896-.735 56.652 15.618 80.36 16 14.596.235 38.53-3.61 58.222-7.625 8.712-1.776 16.05-3.47 22.18-4.91-16.61-19.392-47.196-42.19-83.774-60.38-39.91-19.846-86.81-34.618-129.94-36.798zm-97.768 109.66c-17.693.86-35.45 8.61-51.22 16.005-9.012 4.226-17.343 8.447-24.168 11.486C86.85 469.48 81.11 471 80 471c-25.66 0-48.943-12.707-62-21.492v21.472C33.352 479.837 55.207 489 80 489c7.268 0 13.51-2.78 20.998-6.115 7.488-3.335 15.8-7.56 24.488-11.633 17.376-8.147 36.382-15.234 49.875-14.275 8.73.62 17.46 6.266 27.45 13.51 9.993 7.246 21.062 16.013 35.75 18.396 21.05 3.416 40.977-2.01 59.72-7.215 18.745-5.204 36.403-10.194 52.91-8.705 6 .54 11.362 3.603 18.867 7.564C377.562 484.487 387.252 489 400 489c14.94 0 38.64-4.13 59.537-8.164 15.083-2.91 28.2-5.772 34.463-7.166v-18.39l-.012-.05c0-.002-17.313 3.968-37.863 7.934C435.575 467.13 411.277 471 400 471c-8.694 0-14.606-2.73-21.54-6.39-6.936-3.66-14.852-8.6-25.65-9.573-21.053-1.898-40.784 4.134-59.343 9.287-18.56 5.153-35.852 9.418-52.026 6.793-8.856-1.437-17.89-7.824-28.063-15.2-10.174-7.378-21.676-15.823-36.738-16.894-2.6-.177-5.16-.19-7.576-.074z"
              />
            </svg>
            <span>
              Island
            </span>
          </div>
          <div
            data-testid="category-card-City"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M217 25v30h30V25h-30zm48 0v30h30V25h-30zm48 0v30h30V25h-30zm-96 48v46h30V73h-30zm48 0v46h30V73h-30zm48 0v46h30V73h-30zm-96 64v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zm174 6.508l-94 53.715V215h94v-71.492zM217 201v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zm80 32v110h94V233h-94zM39 256v23h18v-23H39zm178 9v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zM37.562 297l-7 14h146.875l-7-14H37.563zM25 329v158h23v-23h32v23h48v-23h32v23h23V329H25zm192 0v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zM48 352h32v16H48v-16zm80 0h32v16h-32v-16zm265 9v126h31v-23h32v23h31V361h-94zM48 384h32v16H48v-16zm80 0h32v16h-32v-16zm89 9v46h30v-46h-30zm48 0v46h30v-46h-30zm48 0v46h30v-46h-30zM48 416h32v16H48v-16zm80 0h32v16h-32v-16zm89 41v30h30v-30h-30zm48 0v30h30v-30h-30zm48 0v30h30v-30h-30z"
              />
            </svg>
            <span>
              City
            </span>
          </div>
          <div
            data-testid="category-card-Hotel"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="7.5"
                cy="11.5"
                r="2.5"
              />
              <path
                d="M17.205 7H12a1 1 0 0 0-1 1v7H4V6H2v14h2v-3h16v3h2v-8.205A4.8 4.8 0 0 0 17.205 7zM13 15V9h4.205A2.798 2.798 0 0 1 20 11.795V15h-7z"
              />
            </svg>
            <span>
              Hotel
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
